File: 11-ContainerWithMostWater/11-ContainerWithMostWater.cpp
// Last updated: 03/06/2025, 09:55:20
class Solution {
public:
    int maxArea(vector<int>& height) 
    {
        int l = 0, r = height.size() - 1, ans = -1;
        while(l<r){
            int temp = min(height[l], height[r]);
            int water = abs((l-r)*temp);
            ans = max(ans,water);
            if (height[l] < height[r]) l++;
            else r--;
        }
        return ans;

    }
};

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.java
// Last updated: 03/06/2025, 09:55:15
class Solution {
    public int maxArea(int[] arr) {
        int l = 0;
        int r = arr.length -1 ;
        int w = 0;
        int mv = 0;
        while ( l < r ){
            w = r - l;
            int h =  Math.min (arr [l] , arr[r]);
            int v = w * h;
            mv = Math.max(v, mv);
             // Move l forward while arr[l] <= h
            while (l < r && arr[l] <= h) l++;
            // Move r backward while arr[r] <= h
            while (l < r && arr[r] <= h) r--;
        }
        return mv;
    }
}